### ISSUE TITLE
Sporadically failing tests with Poltergeist

### ISSUE THREAD

**Issue by michaelrkn** [travis-ci/travis-ci#1708]
I'm running into a similar issue with https://github.com/epicodus/textbook. All my tests pass fine on my dev machine, and I even provisioned a Nitrous.io box and all my tests pass there as well. But I get weird intermittent failures on Travis CI, which I think all relate to Poltergeist/PhantomJS. Here are my last four builds: https://travis-ci.org/epicodus/textbook/builds/14757934, https://travis-ci.org/epicodus/textbook/builds/14758068, https://travis-ci.org/epicodus/textbook/builds/14770897, https://travis-ci.org/epicodus/textbook/builds/14771361. They all are the exact same codebase, with the exception that the last one includes `debug: true`, but I get different Poltergeist-related errors each time (and a strange `undefined method 'map' for nil:NilClass` occasionally). I've tried raising raising the timeout with `Capybara.default_wait_time = 5` to no avail.

**Comment by jhaesus** [comment]
Why close something that is not fixed? It appears to only be a problem on travis-ci.

**Comment by BanzaiMan** [comment]
@jhaesus Are you seeing the problem now? We are happy to spin up a debug VM for you. Please email support@travis-ci.com.

**Comment by sporkmonger** [comment]
@BanzaiMan We are currently seeing either this problem or something very similar, yes. I've got a few more debug tricks up my sleeves, but if those fail I'll open a support request.

**Comment by metaskills** [comment]
I have found that in many cases I can help my integration tests be more deterministic by taking advantage of Poltergeist's blacklist feature. In my case, I have blacklisted host names like the following.

```
typekit.net
facebook.net
facebook.com
google.com
google-analytics.com
...
```

The idea is to turn off anything that a) is not really needed for testing and b) could affect the page load completion under the CI environment. I have noticed this helps tremendously. However, I still get a few random failures that I can never reproduce locally under load. So I think there is still something afoot. HTH in the meantime.
